holopy/core/metadata.py

```diff
@@ -134,22 +142,45 @@ def from_flat(a):
         return a.unstack('flat')
     return a
 
-def sphere_coords(a, origin=(0,0,0), wavevec=1, include_r=True):
-    if hasattr(a,'theta') and hasattr(a, 'phi') and not include_r:
-        # we are working in far field
-        return {'theta': a.theta, 'phi': a.phi}
+def sphere_coords(a, origin=(0,0,0), wavevec=1):
+    if hasattr(a,'theta') and hasattr(a, 'phi'):
+        out = {'theta': a.theta.values, 'phi': a.phi.values, 'point':a.point.values}
+        if hasattr(a, 'r') and any(a.r < np.inf):
```

Comment:
Probably consider using np.isfinite
https://docs.scipy.org/doc/numpy/reference/generated/numpy.isfinite.html
floating point infinities can be tricky beasts sometimes so the numpy function is probably more reliably correct.